swap_meet/vendor.py

```diff
@@ -1,2 +1,196 @@
+
+        Parameters
+        ----------
+        inventory (list, optional): List of items, default is a empty list.
+        """
+
+        self.inventory = inventory if inventory else []
+
+
+    def add(self, item):
+        """
+        Adds an item to the inventory.
+        
+        Parameters
+        ----------
+        item (str): an item to add
+
+        Returns
+        -------
+        item
+        """
+
+        self.inventory.append(item)
+        return item
+
+
+    def remove(self, item):
+        """
+        Removes an item from the inventory.
+        
+        Parameters
+        ----------
+        item(str): an item to remove
+
+        Returns
+        -------
+        Returns item if successful, False otherwise.
+        """
+
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+
+    def get_by_category(self, category):
+        """
+        Returns a list of items with same category as argument.
+        
+        Parameters
+        ----------
+        category(str): category of an inventory item
+        
+        Returns
+        -------
+        list
+        """
+
+        items_with__same_category_list = []
+        for item in self.inventory:
+            if category in item.category:
+                items_with__same_category_list.append(item) 
+        return items_with__same_category_list
+
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        """
+        Swaps specific item (my_item) with another vendor's item (their_item).
+        
+        Parameters
+        ----------
+        other_vendor(object): the other vendor that this vendor is trading with
+        my_item(object): an object in this vendor's inventory
+        their_item(object): an object in the other vendor's inventory
+        
+        Returns
+        -------
+        Returns True if the trade successful, False otherwise
+        """
+        if not my_item in self.inventory or not their_item in other_vendor.inventory:
+            return False
+        
+        other_vendor.inventory.append(my_item)
+        self.inventory.remove(my_item)
+
+        self.inventory.append(their_item)
+        other_vendor.inventory.remove(their_item)
+        return True
+
+    
+    def swap_first_item(self, other_vendor):        
+        """
+        Swaps the first item in each inventory between this vendor and the other vendor.
+        
+        Parameters
+        ----------
+        other_vendor(object): the other vendor that this vendor is trading with
+        
+        Returns
+        -------
+        Returns True if the trade was successful, False otherwise
+        """
+        
+        if not self.inventory or not other_vendor.inventory:
+            return False
+            
+        self.swap_items(other_vendor,self.inventory[0],other_vendor.inventory[0])
+        return True
```

Comment:
👍

Reply:
Currently, swapping any items has a time complexity of O(n) due to the `remove` method in `swap_items`. We can make a subtle improvement to O(1) by using multiple-assignment instead:

```
        self.inventory[0], vendor.inventory[0] = vendor.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple-assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp